packages/api/portfolioDrafts/funding/createFundingStep.ts

```diff
@@ -1,59 +1,98 @@
-import { UpdateCommand } from "@aws-sdk/lib-dynamodb";
 import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
-import { FundingStep } from "../../models/FundingStep";
+import { ApiSuccessResponse, SuccessStatusCode, ValidationErrorResponse } from "../../utils/response";
+import { dynamodbDocumentClient as client } from "../../utils/dynamodb";
 import { FUNDING_STEP } from "../../models/PortfolioDraft";
-import { dynamodbClient as client } from "../../utils/dynamodb";
-import { DATABASE_ERROR, NO_SUCH_PORTFOLIO_DRAFT, REQUEST_BODY_EMPTY, REQUEST_BODY_INVALID } from "../../utils/errors";
-import { ApiSuccessResponse, SuccessStatusCode } from "../../utils/response";
-import { isFundingStep, isValidJson } from "../../utils/validation";
+import { FundingStep } from "../../models/FundingStep";
+import { UpdateCommand } from "@aws-sdk/lib-dynamodb";
+import {
+  DATABASE_ERROR,
+  NO_SUCH_PORTFOLIO_DRAFT,
+  REQUEST_BODY_EMPTY,
+  REQUEST_BODY_INVALID,
+  PATH_PARAMETER_REQUIRED_BUT_MISSING,
+} from "../../utils/errors";
+import {
+  isFundingStep,
+  isValidJson,
+  isValidDate,
+  isPathParameterPresent,
+  isClin,
+  isValidUuidV4,
+} from "../../utils/validation";
+import { ErrorCodes } from "../../models/Error";
+
+/**
+ * Validation error tuple
+ * verror = ["clin_number", "param_name", "param_value", "message"];
+ */
+export let verror: [string, string, string, string];
```

Comment:
Along these lines, what are the best practices for sharing test code? It gets confusing easily - you don't want to put it in a `test` folder or name it `.test.ts` because that implies that it's literally testing something when in fact it's really "main" code that just doesn't ship with the product. The preferred solution in Java is to create a stand-alone jar which is then  depended upon at the `test` scope. But I don't see an obvious solution for us here that doesn't involve building our own modules, which feels like overkill for now.

And getting back to the task at hand - do we really want to `export` anything from a single unit test?

Reply:
`ClinValidationError` added to replace verror tuple in 13a0ea6903f2595a62d33296dfbca77f08c3ae83.